### PR TITLE
Implement sparse pattern GCN model

### PR DESCRIPTION
**Pull request by nd-02110114**
I implemented GCN model with adj list. The input adj list shape is (2, E).

### CODE REVIEW

File: examples/tox21/gcn_sparse_pattern_example.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    early_stop = EarlyStopping(patience=early_stop_patience)
+    init_batch, init_graph_idx = collate_fn(
+        next(train_dataset.iterbatches(batch_size=batch_size)), task_index
+    )
+    params, state = model.init(next(rng_seq), *init_batch[0], init_graph_idx, True)
+    opt_state = optimizer.init(params)
+    for epoch in range(num_epochs):
+        # train
+        start_time = time.time()
+        # FIXME : This for loop should be rewrited by lax.scan or lax.fori_loop
```
Comment (nd-02110114) [inline examples/tox21/gcn_sparse_pattern_example.py:154]: Please confirm the details from this issue : https://github.com/google/jax/issues/2242
SparseCGN uses `jax.ops.index_add`, but a large Python "for" loop leads to a serious performance issue when using `jax.ops.index_add` (training time of this example is 24 times than the example of PadGCN)

Currently, I try to fix this problem.

**Comment by nd-02110114** [comment]
@rbharath 
I finish implementing SparseGCN model and example. 
I will make another PR for fixing a performance issue.

File: examples/tox21/gcn_sparse_pattern_example.py (hunk 1 above)
Comment (rbharath) [reply examples/tox21/gcn_sparse_pattern_example.py:154]: Good find. Maxes sense!

**Comment by rbharath** [comment]
Sg! Feel free to merge this in when ready

**Pull request merged**